java/src/main/java/genericpubsub/ManagedSubscribe.java

```diff
@@ -230,50 +234,30 @@ public synchronized void close() {
      */
     private synchronized void abort(Exception e) {
         serverStream.onError(e);
-        isActive = false;
+        isActive.set(false);
         this.notifyAll();
     }
 
     /**
-     *  function illustrates the statues of the stream on a fixed intervals.
-     *  This is for demonstration purposes only, and not to be used in a production setting.
+     * Helper function to halt the current thread.
      */
-    public synchronized boolean waitForEvents(int numEvents) {
```

Comment:
We determined that we want a long-running example so we shouldn't need `waitForEvents` and `waitForEventsWithMaxTimeOut` - these functions are intended to be used for a subscriber that closes once a certain number of events is received (or a timeout is reached)